tool/planet/agent.go

```diff
@@ -232,27 +254,95 @@ var electedUnits = []string{
 	"kube-apiserver.service",
 }
 
-func unitsCommand(command string) error {
-	log.Debugf("executing %v on %v", command, electedUnits)
-	var errors []error
-	for _, unit := range electedUnits {
-		cmd := exec.Command("/bin/systemctl", command, unit)
-		log.Debugf("executing %v", cmd)
-		out, err := cmd.CombinedOutput()
-		if err != nil {
-			errors = append(errors, err)
-			// Instead of failing immediately, complete start of other units
-			log.Warningf("failed to execute %v: %s\n%v", cmd, out, trace.DebugReport(err))
+func newUnitMonitor() (*unitMonitor, error) {
+	conn, err := systemdDbus.New()
```

Comment:
So I don't know how this library works, but the potential source of new problems I see here is - could this connection somehow "go bad"? Which would effectively make this agent to be unable to stop/start units. Before, we just used systemctl directly so these kinds of issues were abstracted from us. Using cli directly is not the most pretty way but I'm just afraid this approach might reduce stability instead of improving it.